src/tests/ftest/slurm_setup.py

```diff
@@ -0,0 +1,281 @@
+#!/usr/bin/python2
+"""
+  (C) Copyright 2018-2019 Intel Corporation.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+  GOVERNMENT LICENSE RIGHTS-OPEN SOURCE SOFTWARE
+  The Government's rights to use, modify, reproduce, release, perform, display,
+  or disclose this software are subject to the terms of the Apache License as
+  provided in Contract No. B609815.
+  Any reproduction of computer software, computer software documentation, or
+  portions thereof marked with this legend must also reproduce the markings.
+"""
+
+import socket
+import argparse
+import logging
+import os
+import pwd
+import re
+from general_utils import pcmd, run_task
+from ClusterShell.NodeSet import NodeSet
+
+slurm_conf = "/etc/slurm/slurm.conf"
+slurmdbd_conf = "/etc/slurm/slurmdbd.conf"
+
+package_list = ["munge-devel", "munge-libs",
+                "readline-devel", "perl-ExtUtils-MakeMaker",
+                "openssl-devel", "pam-devel", "rpm-build",
+                "perl-DBI", "perl-Switch", "munge", "mariadb-devel",
+                "slurm", "slurm-contribs", "slurm-devel",
+                "slurm-example-configs", "slurm-libpmi", "slurm-openlava",
+                "slurm-pam_slurm", "slurm-perlapi", " slurm-slurmctld",
+                "slurm-slurmd", "slurm-slurmdbd", "slurm-torque",
+                "mariadb-server"]
```

Comment:
Why are all of these `*-devel` packages needed?  `-devel` packages are typically include files and linker libraries used for building software.